### PR TITLE
Change example to newest development

### PR DESCRIPTION
**Pull request by gjohansson-ST**

### CODE REVIEW

File: custom_components/example_sensor/sensor.py [hunk 1]
```diff
@@ -24,25 +24,12 @@ class ExampleSensor(SensorEntity):
...
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def unit_of_measurement(self) -> str:
-        """Return the unit of measurement."""
-        return TEMP_CELSIUS
+        self._attr_name = 'Example Temperature'
+        self._attr_native_unit_of_measurement = TEMP_CELSIUS
```
Comment (frenck) [inline custom_components/example_sensor/sensor.py]: These are static, so should be implemented as class attributes

File: custom_components/example_sensor/sensor.py [hunk 2]
```diff
@@ -24,25 +24,12 @@ class ExampleSensor(SensorEntity):
     def __init__(self):
         """Initialize the sensor."""
         self._state = None
```
Comment (ludeeus) [inline custom_components/example_sensor/sensor.py]: The _state attribute is no longer used

Comment (gjohansson-ST) [reply custom_components/example_sensor/sensor.py]: Just changed

File: custom_components/example_sensor/sensor.py [hunk 3]
```diff
@@ -21,28 +25,17 @@ def setup_platform(
...
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def unit_of_measurement(self) -> str:
-        """Return the unit of measurement."""
-        return TEMP_CELSIUS
+    def __init__(self) -> None:
+        """Initialize the sensor."""
```
Comment (ludeeus) [inline custom_components/example_sensor/sensor.py]:
```suggestion
```
The init is not used, and can be removed

Comment (gjohansson-ST) [reply custom_components/example_sensor/sensor.py]: I was thinking to put something there just to maintain it as it's an example but could not figure out something good...

Comment (gjohansson-ST) [reply custom_components/example_sensor/sensor.py]: Ok. I went with it and removed __init__

**Review by ludeeus — APPROVED** [review]
Thanks 👍

**Pull request merged**